### DISCUSSION TITLE
[help] . ./activate does not exist

### DISCUSSION THREAD

**Discussion by altered-beast** [Chia-Network/chia-blockchain#3974] [Support]
**Describe the issue**
when installing chia from cli i get the error ``-bash: ./activate: No such file or directory

**To Reproduce**
follow the install instructions for ubuntu 20.04

**Expected behavior**
i expect for . ./activate to put me into a venv and allow me to run chia commands

**Additional context**
tested on a another similar machine with same results

**Comment by TimFoerster** [comment]
Are you in the correct folder and have the correct git branch?

what does `ls` gives you?

**Reply by room101-dev** [reply]
The install scripts have been broken for months, 

I'm on linux and I have to view the scripts cut&paste out the relevant lines for linux, and then run my own batch file

Especially bad is the install-gui.sh file, its complete crap.

IMHO the scripts are written by a guy on a MAC, and tested there and that's it, they don't work anywhere else.

...

Then there is the issue of no 'requirements.txt' for the python, and of course chia-net has no idea of what version of npm & nodejs to run

...

Months ago I could install just using the install-wiki, linux section, since after 1.3 nothing installs anymore, I have test 18 & 20 ubuntu both fail, I don't care anymore I have quit using the GUI, and on the blockchain I run 1.3 for my chia-pos plotting, everything is cmd-line. My original install is on linux-mint 19, its still stable I used to update daily, and I quit updating it weeks ago.

The only thing that makes sense anymore is to 'plot' with 1.3, and generate  plots and mine them on hpool(dot-com), and earn some xch, pay no attention to what chia-net (dev) are doing, because its a lot like watching sausage being made, you don't want to know, or watch.

I have one machine running version 1.5 with 'chia start -r all' running, which is essentially a full node, I can use the cmd-line  just like bitcoin-cli and find out anything I want to know. If I want to know a balance I would use the chia-explorer, because by defn they  must be synced, even my original linux-mint chia main-net dropped sync hourly for 20minutes, I have never in 3 months seen a single stable version. It could just be the GUI failing to correctly read the logs, I just don't want to waste anymore time on chia-net main-net; Like I have said, gen plots earn chia, all else is BS.

**Comment by altered-beast** [comment]
i  am in the correct folder and ls gives this 

`azure-pipelines.yml  BUILD_TIMELORD.md  chia                 CODE_OF_CONDUCT.md  install-gui.sh  install.sh           LICENSE     mypy.ini        README.md        setup.py
build_scripts        CHANGELOG.md       chia-blockchain-gui  CONTRIBUTING.md     INSTALL.md      install-timelord.sh  mozilla-ca  pyproject.toml  run-py-tests.sh  tests`

**Comment by TimFoerster** [comment]
did u run `sh install.sh` before?

**Comment by altered-beast** [comment]
yes and it finished with no errors

**Comment by moshenskyDV** [comment]
Could you please provide full `sh install.sh` log?

**Reply by altered-beast** [reply]
`tvvmh@tvvmh:~/chia-blockchain$ sh install.sh
apt-get is /usr/bin/apt-get
Installing on Ubuntu/Debian 20.04 LTS
[sudo] password for tvvmh:
Hit:1 http://ppa.launchpad.net/ondrej/php/ubuntu focal InRelease
Hit:2 http://apt.sonarr.tv master InRelease
Hit:3 https://download.docker.com/linux/ubuntu focal InRelease
Hit:4 https://deb.nodesource.com/node_12.x focal InRelease
Hit:5 http://gb.archive.ubuntu.com/ubuntu focal InRelease
Get:6 http://gb.archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:7 http://security.ubuntu.com/ubuntu focal-security InRelease [109 kB]
Get:8 http://gb.archive.ubuntu.com/ubuntu focal-backports InRelease [101 kB]
Hit:9 http://repo.zabbix.com/zabbix/5.0/ubuntu focal InRelease
Get:11 https://repo.jellyfin.org/ubuntu focal InRelease [6,636 B]
Ign:13 http://linux.dell.com/repo/community/openmanage/930/focal focal InRelease
Hit:12 https://dl.ubnt.com/unifi/debian stable InRelease
Get:14 http://gb.archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [955 kB]
Hit:15 http://linux.dell.com/repo/community/openmanage/950/focal focal InRelease
Get:16 http://gb.archive.ubuntu.com/ubuntu focal-updates/main i386 Packages [466 kB]
Get:17 http://gb.archive.ubuntu.com/ubuntu focal-updates/main Translation-en [218 kB]
Get:18 http://gb.archive.ubuntu.com/ubuntu focal-updates/main amd64 DEP-11 Metadata [265 kB]
Get:19 http://gb.archive.ubuntu.com/ubuntu focal-updates/main DEP-11 48x48 Icons [55.6 kB]
Get:20 http://gb.archive.ubuntu.com/ubuntu focal-updates/main DEP-11 64x64 Icons [87.9 kB]
Get:21 http://security.ubuntu.com/ubuntu focal-security/main i386 Packages [224 kB]
Get:22 http://gb.archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata [13.2 kB]
Get:23 http://gb.archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [768 kB]
Get:24 http://gb.archive.ubuntu.com/ubuntu focal-updates/universe i386 Packages [567 kB]
Err:25 http://linux.dell.com/repo/community/openmanage/930/focal focal Release
  404  Not Found [IP: 143.166.156.113 80]
Get:26 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [629 kB]
Get:27 http://gb.archive.ubuntu.com/ubuntu focal-updates/universe Translation-en [165 kB]
Get:28 http://gb.archive.ubuntu.com/ubuntu focal-updates/universe amd64 DEP-11 Metadata [303 kB]
Get:29 http://gb.archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata [17.2 kB]
Get:30 http://gb.archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 DEP-11 Metadata [2,468 B]
Get:31 http://security.ubuntu.com/ubuntu focal-security/main amd64 DEP-11 Metadata [24.4 kB]
Get:32 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [558 kB]
Get:33 http://gb.archive.ubuntu.com/ubuntu focal-backports/universe amd64 DEP-11 Metadata [1,768 B]
Get:34 http://security.ubuntu.com/ubuntu focal-security/universe i386 Packages [441 kB]
Get:35 http://security.ubuntu.com/ubuntu focal-security/universe amd64 DEP-11 Metadata [58.3 kB]
Get:36 http://security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata [10.8 kB]
Hit:10 https://packagecloud.io/pufferpanel/pufferpanel/ubuntu focal InRelease
Reading package lists... Done
E: The repository 'http://linux.dell.com/repo/community/openmanage/930/focal focal Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
N: Skipping acquisition of configured file 'main/binary-i386/Packages', as repository 'http://repo.zabbix.com/zabbix/5.0/ubuntu focal InRelease' doesn't support architecture 'i386'`

that should be it

**Comment by moshenskyDV** [comment]
> yes and it finished with no errors

That's not true. You have error during `apt-get update`. I suggest you to open `install.sh` and remove line40 or 44 or 48 (depending on your OS distribution). Save the file and run `sh install.sh` again. And it should be done correctly.
I had the same error. So if it mass, I plan to create PR to repository to fix it. Right now please use workaround I described.

**Reply by altered-beast** [reply]
ah sorry i must of missed that ill try the work around and get backto you

**Reply by altered-beast** [reply]
yep that fixed it thanks for all the help !  :)

**Reply by moshenskyDV** [reply]
@altered-beast please mark that answer as "response" to close the issue, and please finger up for #3990 PR (that fixes this issue)

**Reply by mdsutton81** [reply]
"Right now please use workaround I described."  What is the work around?  I do not see it mentioned here?  You say remove line 40, or 44 or 48?  What is the content of these lines?  For me line 40 #some comment, 44 is #Ubuntu 48 is some "Elif"  I doubt these are what you are referencing.  But am stumped at what your "Work around" is??  Thank you.

**Reply by igbenic** [reply]
@mdsutton81 "apt-get update" failed, so comment out the "sudo apt-get update" line for your distribution

**Comment by jakobcoola** [comment]
Hi! i have the same issue... and tried to remove line 48 (no result) line 44 and 40 is just a comment. @moshenskyDV 
Any idea?

**Comment by ozz-life** [comment]
A similar problem. Debian 10 :(

**Comment by BrandtH22** [comment]
Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
